cg/store/api/base.py

```diff
@@ -127,3 +131,77 @@ def _get_latest_analyses_for_cases_query(self) -> Query:
                 self.Analysis.started_at == case_and_date_subquery.c.started_at,
             ),
         )
+
+    def _get_filtered_case_query(
```

Comment:
Great work on this one. We should note this function down for a major refactoring in the future

Reply:
Agreed!